Kudu.Services/Deployment/PushDeploymentController.cs

```diff
@@ -138,6 +140,164 @@ public class PushDeploymentController : ApiController
+                            return response;
+                        }
+                        if (string.IsNullOrWhiteSpace(deploymentInfo.TargetPath))
+                        {
+                            deploymentInfo.FileName = "app.jar";
+                        }
+                        break;
+                    case "lib":
+                        if (string.IsNullOrWhiteSpace(deploymentInfo.TargetPath))
+                        {
+                            response.StatusCode = HttpStatusCode.BadRequest;
+                            response.Content = new StringContent("Path must be defined for library file deployments");
+                            return response;
+                        }
+                        else
+                        {
+                            deploymentInfo.FileName = Path.GetFileName(path);
+                            deploymentInfo.TargetPath = Path.GetDirectoryName(path);
+                        }
+                        break;
+                    case "startup":
+                        //assumes windows application for Kudu deploymnets
+                        if (string.IsNullOrWhiteSpace(deploymentInfo.TargetPath))
+                        {
+                            //deploymentInfo.TargetPath = Path.Combine(_environment.RootPath, "startup.bat");
+                            deploymentInfo.TargetPath = _environment.RootPath;
+                            deploymentInfo.FileName = "startup.bat";
+                        }
+                        break;
+                    case "ear":
+                        //currently not supported on Windows but here for future use
+                        if (website_framework != "javeee")
+                        {
+
+                            response.StatusCode = HttpStatusCode.BadRequest;
+                            response.Content = new StringContent("EAR files cannot be deployed to " + website_framework + " web apps.");
+                            return response;
+                        }
+                        if (string.IsNullOrWhiteSpace(deploymentInfo.TargetPath))
+                        {
+                            deploymentInfo.FileName = "app.ear";
+                        }
+                        break;
+                    case "static":
+                        if (string.IsNullOrWhiteSpace(path))
+                        {
+                            response.StatusCode = HttpStatusCode.BadRequest;
+                            response.Content = new StringContent("Path must be defined for static file deployments");
+                            return response;
+                        }
+                        else
+                        {
+                            deploymentInfo.FileName = Path.GetFileName(path);
+                            deploymentInfo.TargetPath = Path.GetDirectoryName(path);
```

Comment:
Can you add a comment (in code) about how targetpath is used and how filename is used?

Reply:
When to use FileName versus when to use TargetPath is not clear. Can you share some details?

---
In reply to: [456028286](https://github.com/shrishrirang/kudu/pull/2#discussion_r456028286) [](ancestors = 456028286)

Reply:
Can you share details and add it to the code as a comment?

---
In reply to: [456719509](https://github.com/shrishrirang/kudu/pull/2#discussion_r456719509) [](ancestors = 456719509,456028286)